tests/examples/wibox/awidget/prompt/keypress.lua

```diff
@@ -0,0 +1,33 @@
+local parent    = ... --DOC_NO_USAGE --DOC_HIDE
+local wibox     = require( "wibox"     ) --DOC_HIDE
+local awful     = { prompt = require("awful.prompt"),--DOC_HIDE
+                    util   = require("awful.util"),--DOC_HIDE
+                    screen = require("awful.screen")}--DOC_HIDE
+local beautiful = require( "beautiful" ) --DOC_HIDE
+local naughty   = {} --DOC_HIDE
+
+    local atextbox = wibox.widget.textbox()
+
+    local notif = nil
+    awful.prompt.run {
+        prompt               = "<b>Run: </b>",
+        keypressed_callback  = function(mod, key, cmd) --luacheck: no unused args
+            if key == "Shift_L" then
+                notif = naughty.notify { text = "Shift pressed" }
+            end
+        end,
+        keyreleased_callback = function(mod, key, cmd) --luacheck: no unused args
+            if notif then
+                notif.destroy_cb(naughty.notificationClosedReason.undefined)
```

Comment:
Are you trying to destroy the notification? That's not how it works. This only called the `destroy = foo`-callback that you gave to `naughty.notify`, so should be `nil` here.

I think you want `naughty.destroy(notify)` (the reason argument can be nil).